### PR TITLE
Issue #9 layout application architecture

### PR DESCRIPTION
**Pull request by wagnet**

### CODE REVIEW

**Comment by kirons323** [comment]
LGTM

File: CharacterData.py [hunk 1]
```diff
@@ -1,112 +1,190 @@
...
-			self.rates['Str'] = int(array[3])
-			self.rates['Mag'] = int(array[4])
-			self.rates['Skl'] = int(array[5])
-			self.rates['Spd'] = int(array[6])
-			self.rates['Lck'] = int(array[7])
-			self.rates['Def'] = int(array[8])
-			self.rates['Res'] = int(array[9])
+			"""Inits CharacterData with name, game_class_options and 
+			growth_rate_class"""
+			self.name = array[0]
```
Comment (Asudano) [inline CharacterData.py:141]: If we are going to have getters in this class that we should have these as private variables

File: CharacterData.py [hunk 2]
```diff
@@ -1,112 +1,190 @@
...
+			self.base_stats['Lck'] = int(array[8])
+			self.base_stats['Def'] = int(array[9])
+			self.base_stats['Res'] = int(array[10])
+			pass	
+			
+		def addClassAndGrowthRates(self, array):
+			self.game_class_options.append(array[1])
+			gr = self.GrowthRates(array)
+			self.growth_rate_class[array[1]] = gr		
+			pass
```
Comment (Asudano) [inline CharacterData.py]: I am guessing you meant to get rid of this pass

File: CharacterData.py [hunk 3]
```diff
@@ -1,112 +1,190 @@
...
+			array[10] = array[10].strip()
+			self.base_stats['HP'] = int(array[3])
+			self.base_stats['Str'] = int(array[4])
+			self.base_stats['Mag'] = int(array[5])
+			self.base_stats['Skl'] = int(array[6])
+			self.base_stats['Spd'] = int(array[7])
+			self.base_stats['Lck'] = int(array[8])
+			self.base_stats['Def'] = int(array[9])
+			self.base_stats['Res'] = int(array[10])
+			pass
```
Comment (Asudano) [inline CharacterData.py]: Same thing as below

File: CharacterData.py [hunk 4]
```diff
@@ -1,112 +1,190 @@
...
-			self.rates['Def'] = int(array[8])
-			self.rates['Res'] = int(array[9])
+			"""Inits CharacterData with name, game_class_options and 
+			growth_rate_class"""
+			self.name = array[0]
+			self.game_class_options = []
+			self.growth_rate_class = {}
+			self.base_stats = {}
+			self.base_class = ""
+			self.addClassAndGrowthRates(array)
```
Comment (Asudano) [inline CharacterData.py]: This should be a private class

File: CharacterData.py [hunk 5]
```diff
@@ -1,112 +1,190 @@
...
+			level : an int representing an in game level
+			game_class: a str describing an in game class
+
+		Returns:
+			A state object representing the average for this character at the
+				given class and level.
+		"""
+		pass
+
+	if False:
```
Comment (Asudano) [inline CharacterData.py:124]: Why is this here?

File: CharacterData.py [hunk 6]
```diff
@@ -1,112 +1,190 @@
...
+				self.rates = {}
+				self.rates['HP'] = int(array[2])
+				self.rates['Str'] = int(array[3])
+				self.rates['Mag'] = int(array[4])
+				self.rates['Skl'] = int(array[5])
+				self.rates['Spd'] = int(array[6])
+				self.rates['Lck'] = int(array[7])
+				self.rates['Def'] = int(array[8])
+				self.rates['Res'] = int(array[9])
+				pass
```
Comment (Asudano) [inline CharacterData.py]: Guessing you didn't mean to leave this

File: CharacterData.py [hunk 7]
```diff
@@ -1,112 +1,190 @@
...
+			Returns:
+				GrowthRates object
+			"""
+			pass
+
+		@property
+		def rates(self):
+			return self.__rates
+
+		if False:
```
Comment (Asudano) [inline CharacterData.py:44]: Guessing you will be changing this

File: Reader.py [hunk 8]
```diff
@@ -1,13 +1,27 @@
...
 	state = 0
 	game_characters = [] # list of characters and their game data
 	max_stats = {} # nested dictionary {class: {stat name: stat}}
 	promotion_gains = {} # nested dictionaries {base_class: {promoted_class: {stat name: {stat gains/losses}}}}
 	base_classes = [] # list of base classes
 	promoted_classes = [] # list of promoted classes
 	
+	try:
+		infile = open(filepath, "r")
+	except:
```
Comment (Asudano) [inline Reader.py]: It is not good practice to have a general except statement. Please fill in the specific case you will be excepting

File: CharacterData.py (hunk 1 above)
Comment (wagnet) [reply CharacterData.py:141]: Wrapped in an "if False" block. To be replaced by private variable __name described in "Attributes" portion of doc_string at the top of file. Resolved

File: CharacterData.py (hunk 3 above)
Comment (wagnet) [inline CharacterData.py]: Wrapped in an "if False" block. To be replaced by a State object defined through "add_initial_state(self, state)" member function.

File: CharacterData.py (hunk 5 above)
Comment (wagnet) [reply CharacterData.py:124]: Encapsulates logic that will likely be re-used to avoid re-implementation

File: CharacterData.py (hunk 6 above)
Comment (wagnet) [inline CharacterData.py]: Logic will need to be re-used later with indices - 2 used instead of current array[] indices.

File: CharacterData.py (hunk 7 above)
Comment (wagnet) [reply CharacterData.py:44]: Correct. Will be removed after logic is used. Same as above.

File: CharacterData.py [hunk 9]
```diff
@@ -1,115 +1,198 @@
...
+			self.base_stats['Lck'] = int(array[8])
+			self.base_stats['Def'] = int(array[9])
+			self.base_stats['Res'] = int(array[10])
+			pass	
+			
+		def add_class_and_growth_rates(self, array):
+			self.game_class_options.append(array[1])
+			gr = self.GrowthRates(array)
+			self.growth_rate_class[array[1]] = gr		
+			pass
```
Comment (Asudano) [inline CharacterData.py]: Prob should get ride of this pass

File: CharacterData.py (hunk 3 above)
Comment (Asudano) [reply CharacterData.py]: I am talking about the pass command
